File: C++ solutions/1877. Minimize Maximum Pair Sum in Array.cpp
class Solution {
public:
    int minPairSum(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int ans=0;
        for(int i=0,j=nums.size()-1;i<j;i++,j--){
            ans=max(ans,nums[i]+nums[j]);
        }

        if(nums.size()%2) ans=max(ans,(nums[nums.size()/2]+nums[(nums.size()/2)+1]));
        return ans;
    }
};

File: JavaScript solutions/1877. Minimize Maximum Pair Sum in Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minPairSum = function (nums) {
  nums.sort((a, b) => a - b);

  let ans = 0;
  for (let i = 0, j = nums.length - 1; i < j; i++, j--) {
    ans = Math.max(ans, nums[i] + nums[j]);
  }

  if (nums.length % 2 === 1) {
    ans = Math.max(
      ans,
      nums[Math.floor(nums.length / 2)] + nums[Math.floor(nums.length / 2) + 1]
    );
  }

  return ans;
};
